--- schema.js	2020-03-26 03:22:35.000000000 -0400
+++ schema.js.new	2020-09-04 12:30:56.244716995 -0400
@@ -214,6 +214,9 @@
       bright: 'road_map',
       desaturated: 'road_map_desaturated',
       dark: 'dark_map'
+    }),
+    datashader: _joi.default.object({
+      url: _joi.default.string()
     })
   }).default(),
   i18n: _joi.default.object({
@@ -226,4 +229,4 @@
 }).default();
 
 exports.default = _default;
-module.exports = exports.default;
\ No newline at end of file
+module.exports = exports.default;
